python/tutorials/10-block-scaled-matmul.py

```diff
@@ -65,6 +65,52 @@
 #  2. https://docs.nvidia.com/cuda/cublas/#d-block-scaling-factors-layout
 #
 
+# Scale preshuffling on AMD GPUs
+#
+# Scales are stored as 8-bit tensors, where each element scales 32 values from the A or B operand tensors.
+# Since MFMA instructions are wave-level instructions, that means that each thread provides a fixed set of operand values to MFMA instructions.
+#
+# For example, in an MFMA instruction with shape 16x16x128:
+# - 4 threads contribute elements along the K dimension.
+# - 16 threads contribute elements along the M or N dimension.
+#
+# From the perspective of the scales tensor, even if the K dimension is stored contiguously in LDS,
+# each thread sees its elements along K dim as strided due to interleaving with other threads.
+# This striding limits the ability to load scale values using vectorized memory access.
+#
+# Our goal is to reorganize the scale tensor so that:
+# 1. Each thread stores the 4 scale values it needs for 4 MFMA ops in contiguous memory.
+# 2. Continuous threads access contiguous memory locations improving global memory coalescing when bypassing LDS,
+#    which is especially beneficial for "skinny" matmuls.
+#
+# We consider two MFMA cases: one with non-K dimension 16, and one with 32.
+# In both, the minimum tile size for preshuffling is 32x32x256.
+# For example, for a 32x256 operand tile, the corresponding scale tensor has shape 32x8,
+# where each scale covers 32 elements along the K dimension.
+#
+# Each thread holds one scale per MFMA operation. We pack the 4 scale values (for 4 different MFMA ops)
+# next to each other in memory.
+#
+# Case 1: mfma_scaled_16x16x128
+#
+# Packing order: mfma_op_0, mfma_op_2, mfma_op_1, mfma_op_3
+#
+#            K = 128       K = 128
+#        +------------+ +------------+
+#    M=16|  MFMA op 0 | |  MFMA op 1 |
+#        +------------+ +------------+
+#    M=16|  MFMA op 2 | |  MFMA op 3 |
+#        +------------+ +------------+
+#
+# Case 2: mfma_scaled_32x32x64
+#
+# Packing order: mfma_op_0, mfma_op_1, mfma_op_2, mfma_op_3
+#
+#            K=64     K=64     K=64     K=64
+#        +--------+ +--------+ +--------+ +--------+
+#    M=32| op 0   | | op 1   | | op 2   | | op 3   |
+#        +--------+ +--------+ +--------+ +--------+
+
```

Comment:
Are we expecting both to be performant? Or should we only "recommend" one of them by saying "In the following, we only demonstrate with 16x16x128 case given that's the recommended performant path."?